dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -1124,6 +1124,25 @@ RtpsUdpDataLink::RtpsReader::process_data_i(const RTPS::DataSubmessage& data,
     WriterInfo& info = wi->second;
     SequenceNumber seq;
     seq.setValue(data.writerSN.high, data.writerSN.low);
+
+    // determines if this is a direct write to this reader
+    bool directWrite = false;
+    bool directWriteNoMatch = true;
```

Comment:
```suggestion
    bool directWriteMatch = false;
```
In general, avoid using a negatively-named boolean variable since it's harder for a reader to follow.